compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -2390,6 +2404,8 @@ private ParserRuleContext getNextRuleForCloseBrace(int nextLookahead) {
             case INTERPOLATION:
                 endContext();
                 return ParserRuleContext.TEMPLATE_MEMBER;
+            case EXPLICIT_NEW_RHS:
+                return ParserRuleContext.OPEN_PARENTHESIS;
```

Comment:
This Should be ARG_LIST_START. Its also matches against the `{` , but it handles the arg-contexts

Reply:
fixed in 401f175